libeos-payg/manager.c

```diff
@@ -1017,10 +1111,51 @@ file_load_cb (GObject      *source_object,
             }
 
           /* Set the expiry time. No other validation is needed on the loaded
-           * number, as the entire range of the type is valid. */
+           * number, as the entire range of the type is valid.
+           * The value stored on disk is in terms of wall clock time, but we
+           * want self->expiry_time_secs to be in terms of CLOCK_BOOTTIME, so
+           * we're migrating here */
           memcpy (expiry_time_secs.u8, data, sizeof (expiry_time_secs.u8));
           set_expiry_time (self, now_secs,
-                           (expiry_time_secs.u64 > now_secs) ? expiry_time_secs.u64 - now_secs : 0);
+                           (expiry_time_secs.u64 > wallclock_now_secs) ? expiry_time_secs.u64 - wallclock_now_secs : 0);
+
+          /* Delete the file since we now use clock-time and expiry-seconds instead */
+          g_file_delete_async (file, G_PRIORITY_DEFAULT, cancellable,
+                               file_load_delete_cb, g_object_ref (task));
+        }
+    }
+  else if (g_file_equal (file, expiry_seconds_file))
+    {
+      if (data_len == 0)
+        {
+          /* No expiry time is set. Expire immediately. */
+          set_expiry_time (self, now_secs, 0);
+        }
+      else
+        {
+          union
+            {
+              guint64 u64;
+              gchar u8[8];
+            } expiry_secs;
+          G_STATIC_ASSERT (sizeof (expiry_secs.u8) == sizeof (self->expiry_time_secs));
```

Comment:
Looks like the code for reading a 64-bit int from a file could be factored out.

Reply:
In -nonfree I used `GData[Input|Output]Stream` which might be worth a look.

Reply:
I ended up going the safe route and using helper functions but otherwise leaving the code untouched.